source/PrintData.cpp

```diff
@@ -0,0 +1,497 @@
+		cout << firingHeat << ',';
+		cout << firingForce << ',';
+
+		double shieldDmg = outfit.ShieldDamage() * fireRate;
+		cout << shieldDmg << ',';
+		double dischargeDmg = outfit.DischargeDamage() * 100. * fireRate;
+		cout << dischargeDmg << ',';
+		double hullDmg = outfit.HullDamage() * fireRate;
+		cout << hullDmg << ',';
+		double corrosionDmg = outfit.CorrosionDamage() * 100. * fireRate;
+		cout << corrosionDmg << ',';
+		double heatDmg = outfit.HeatDamage() * fireRate;
+		cout << heatDmg << ',';
+		double burnDmg = outfit.BurnDamage() * 100. * fireRate;
+		cout << burnDmg << ',';
+		double energyDmg = outfit.EnergyDamage() * fireRate;
+		cout << energyDmg << ',';
+		double ionDmg = outfit.IonDamage() * 100. * fireRate;
+		cout << ionDmg << ',';
+		double slowDmg = outfit.SlowingDamage() * fireRate;
+		cout << slowDmg << ',';
+		double disruptDmg = outfit.DisruptionDamage() * fireRate;
+		cout << disruptDmg << ',';
+		cout << outfit.Piercing() << ',';
+		double fuelDmg = outfit.FuelDamage() * fireRate;
+		cout << fuelDmg << ',';
+		double leakDmg = outfit.LeakDamage() * 100. * fireRate;
+		cout << leakDmg << ',';
+		double hitforce = outfit.HitForce() * fireRate;
+		cout << hitforce << ',';
+
+		cout << outfit.Homing() << ',';
+		double strength = outfit.MissileStrength() + outfit.AntiMissile();
+		cout << strength << ',';
+
+		double damage = outfit.ShieldDamage() + outfit.HullDamage();
+		double deterrence = .12 * damage / outfit.Reload();
+		cout << deterrence << '\n';
+	}
+	cout.flush();
+}
+
+
+
+void PrintData::PrintEngineStats()
+{
+	cout << "name" << '\t' << "cost" << '\t' << "mass" << '\t' << "outfit space" << '\t'
+		<< "engine capacity" << '\t' << "thrust/s" << '\t' << "thrust energy/s" << '\t'
+		<< "thrust heat/s" << '\t' << "turn/s" << '\t' << "turn energy/s" << '\t'
+		<< "turn heat/s" << '\t' << "reverse thrust/s" << '\t' << "reverse energy/s" << '\t'
+		<< "reverse heat/s" << '\n';
+	for(auto &it : GameData::Outfits())
+	{
+		// Ship non-engines.
+		if(it.second.Category() != "Engines")
+			continue;
+
+		const Outfit &outfit = it.second;
+		cout << it.first << ',';
+		cout << outfit.Cost() << ',';
+		cout << outfit.Mass() << ',';
+		cout << outfit.Get("outfit space") << ',';
+		cout << outfit.Get("engine capacity") << ',';
+		cout << outfit.Get("thrust") * 3600. << ',';
+		cout << outfit.Get("thrusting energy") * 60. << ',';
+		cout << outfit.Get("thrusting heat") * 60. << ',';
+		cout << outfit.Get("turn") * 60. << ',';
+		cout << outfit.Get("turning energy") * 60. << ',';
+		cout << outfit.Get("turning heat") * 60. << ',';
+		cout << outfit.Get("reverse thrust") * 3600. << ',';
+		cout << outfit.Get("reverse thrusting energy") * 60. << ',';
+		cout << outfit.Get("reverse thrusting heat") * 60. << '\n';
+	}
+	cout.flush();
+}
+
+
+
+void PrintData::PrintPowerStats()
+{
+	cout << "name" << ',' << "cost" << ',' << "mass" << ',' << "outfit space" << ','
+		<< "energy gen" << ',' << "heat gen" << ',' << "energy cap" << '\n';
```

Comment:
```suggestion
		<< "energy generation" << ',' << "heat generation" << ',' << "energy capacity" << '\n';
```
I'd be fine with reducing these ones, if this was a huge print line but considering it isnt that long, why would we? May as well always display the variables fully, whatever the place we're at, they could even be defined in namespace (but then I guess we'd use a little bit of ressources when it isnt needed, so prob best not to do that if its part of the main executable)